rules-java/api/src/main/java/org/jboss/windup/rules/apps/java/scan/operation/AddClassFileMetadata.java

```diff
@@ -48,83 +46,81 @@ public void perform(GraphRewrite event, EvaluationContext context, JavaClassFile
+                if (absoluteFile.contains(packageAsPath))
+                {
+                    shouldScan = true;
+                    break;
+                }
+                else
+                {
+                    shouldScan = false;
+                }
+            }
+            if (!shouldScan)
+            {
+                return;
+            }
+
+            // we should scan it, so make sure we get the package name from it
             try (FileInputStream fis = new FileInputStream(payload.getFilePath()))
             {
                 final ClassParser parser = new ClassParser(fis, payload.getFilePath());
                 final JavaClass bcelJavaClass = parser.parse();
                 final String packageName = bcelJavaClass.getPackageName();
+
                 final String qualifiedName = bcelJavaClass.getClassName();
 
                 final JavaClassService javaClassService = new JavaClassService(event.getGraphContext());
                 final JavaClassModel javaClassModel = javaClassService.create(qualifiedName);
-                if (javaCfgService.shouldScanPackage(packageName))
-                {
-                    int majorVersion = bcelJavaClass.getMajor();
-                    int minorVersion = bcelJavaClass.getMinor();
-
-                    String simpleName = qualifiedName;
-                    if (packageName != null && !packageName.equals("") && simpleName != null)
-                    {
-                        simpleName = simpleName.substring(packageName.length() + 1);
-                    }
+                int majorVersion = bcelJavaClass.getMajor();
+                int minorVersion = bcelJavaClass.getMinor();
 
-                    payload.setMajorVersion(majorVersion);
-                    payload.setMinorVersion(minorVersion);
-                    payload.setPackageName(packageName);
-
-                    javaClassModel.setSimpleName(simpleName);
-                    javaClassModel.setPackageName(packageName);
-                    javaClassModel.setQualifiedName(qualifiedName);
-                    javaClassModel.setClassFile(payload);
-                    javaClassModel.setPublic(bcelJavaClass.isPublic());
-
-                    final String[] interfaceNames = bcelJavaClass.getInterfaceNames();
-                    if (interfaceNames != null)
-                    {
-                        for (final String interfaceName : interfaceNames)
-                        {
-                            JavaClassModel interfaceModel = javaClassService.getOrCreatePhantom(interfaceName);
-                            javaClassModel.addImplements(interfaceModel);
-                        }
-                    }
+                String simpleName = qualifiedName;
+                if (packageName != null && !packageName.equals("") && simpleName != null)
+                {
+                    simpleName = simpleName.substring(packageName.length() + 1);
```

Comment:
We have Commons Lang.  Any reason not to use StringUtils.removeStart?  It reads more clearly.